ruby_review_week_four.rb

```diff
@@ -2,36 +2,134 @@
+var = "Max"
+  
+=end
+
 # 2. What Ruby method can you use to make a loop from an array?
 
+# array = [1,2,3,4,5]
+
+# array.each do |arr| 
+#   puts arr
+# end
+
 # 3. In Ruby, what are booleans?
 
+# Booleans are true or false values e.g. var = true, var = false
+
 # 4. How do classes relate to objects?
 
+# Classes are like the template (structure) of objects. They create the basic overview of what each object will contain.
+
 # 5. What is an instance variable?
 
+# An instance variable is a variable that is initizled through the creation of a class e.g.
+
+class Max
+  def initialize
+    @name = "Max"
+    @age = 21
+  end
+end
+
 # 6. Choose an everyday item (cup, computer, person) and give it two instance variables (which will be attributes), and one method (to describe a function of the object). Code the initialize method in full (for these attributes only), and the definition line of the instance method (but do not code the entire method).
 
+class Person
+  def initialize(name, age)
+    @name = name
+    @age = age
+  end
+
+  def to_string
+    puts "Hi #{@name}"
+  end
+end
+
 # 7. What is handy about using gems in Ruby?
 
+# Ruby GEMS code from someone else which you can then put into your platform. They're handy to use because you can access their code and program easily through requiring the GEM so you don't have to code it all yourself. It is also good because the code is updated by them and maintained.
+
 # 8. Define a class called fish, and leave it empty.
 
-# 9. How would you use that class to make three fish objects?
+=begin
+  
+class Fish 
+  def initialize
 
+  end
+end
+  
+=end
+# 9. How would you use that class to make three fish objects?
+  # fish_one = Fish.new
+  # fish_two = Fish.new
+  # fish_three = Fish.new
 # 10. Add two attributes to your fish class, name and species, and make them accessible for reading and writing.
 
+class Fish
+  attr_accessor :name, :species
+
+  def initialize(name, species)
+    @name = name
+    @species = species
+  end
+
+end
+
 # 11. 
 
 # a. Create one fish object and store it in a variable, and 
 
+fish_one = Fish.new("Nemo", "Clown Fish")
+
 # b. on a second line, give that fish object’s name attribute the value of ‘Fred’ (assume you have access).
 
+fish_one.name = "Fred"
+puts fish_one.name
+
 # 12. Define a hash called ‘student’. This hash should have two keys, one called name, and the other called age, and assign a string to the name key, and an integer to the age key. 
 
+student = {
+  name: "Max",
+  age: 21
+}
+
 # 13. Write a Ruby method that takes three arguments and returns the sum (total) of those three arguments.
 
+def ruby_method(x,y,z)
+  return x + y + z
+end
+
+puts ruby_method(20,30,40)
+
 # 14. Write a Ruby method that takes an array and a string as arguments, and returns true if the string is in the array, and false if it is not.
 
+# array = [1]
+# def ruby_method_two(array, string)
+
+#   array.each do |arr| 
+#     if arr.is_a?(String)
```

Comment:
this has a few issues, we'll review it in class next week

Reply:
No worries! I think I over complicated it and thought that the question was asking if the array is changed it should pick up the difference but chat next week about it.  :)